### ISSUE TITLE
Implement Focal Loss for Class Imbalance

### ISSUE THREAD

**Issue by nikhilc523** [nikhilc523/mobile-vision-training#17]
**Description:**

Current training uses standard binary cross-entropy loss, which doesn't handle class imbalance well (~18% fall / ~82% non-fall). Implement focal loss to focus training on hard-to-classify examples and improve performance on minority class (falls).

**Background:**
- Current dataset: ~18% fall / ~82% non-fall imbalance
- Standard BCE treats all samples equally
- Focal loss down-weights easy examples, focuses on hard ones
- Improves recall on minority class (falls) without sacrificing precision

**What is Focal Loss?**
Focal loss adds a modulating factor to cross-entropy loss:
```
FL(pt) = -α(1-pt)^γ * log(pt)
```
- α (alpha): Weight for positive class (typically 0.25)
- γ (gamma): Focusing parameter (typically 2.0)
- pt: Model's estimated probability for the correct class

**Tasks:**
- [ ] Install TensorFlow Addons: `pip install tensorflow-addons`
- [ ] Import focal loss: `from tensorflow_addons.losses import SigmoidFocalCrossEntropy`
- [ ] Update `ml/training/lstm_train.py` to use focal loss
- [ ] Add CLI arguments:
  - `--focal-loss` (enable focal loss)
  - `--focal-alpha` (default: 0.25)
  - `--focal-gamma` (default: 2.0)
- [ ] Test on current 17-sample dataset
- [ ] Compare performance with standard BCE
- [ ] Train on full dataset with focal loss
- [ ] Document results and performance comparison
